File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_ChongLou.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_ChongLou implements ITableOperate {
 public final static String TAB_FILE_DATA = "/ChongLou.txt";
 public static final int ID_FIGHT1_0 = 0 ;
public static final int ID_FIGHT1_1 = 1 ;
public static final int ID_FIGHT1_2 = 2 ;
public static final int ID_FIGHT1_3 = 3 ;
public static final int ID_FIGHT1_4 = 4 ;
public static final int ID_FIGHT1_5 = 5 ;
public static final int ID_COPY_DROP_1 = 6 ;
public static final int ID_COPY_NUM_1 = 7 ;
public static final int ID_COPY_DROP_2 = 8 ;
public static final int ID_COPY_NUM_2 = 9 ;
public static final int MAX_RECORD = 10 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private int m_CopyDrop[] = new int[2];
 public int GetCopyDropbyIndex(int idx) {
 if(idx>=0 && idx<2) return m_CopyDrop[idx];
 return -1;
 }

private int m_CopyNum[] = new int[2];
 public int GetCopyNumbyIndex(int idx) {
 if(idx>=0 && idx<2) return m_CopyNum[idx];
 return -1;
 }

private int m_Fight1[] = new int[6];
 public int GetFight1byIndex(int idx) {
 if(idx>=0 && idx<6) return m_Fight1[idx];
 return -1;
 }

public boolean LoadTable(HashMap<Integer, Table_ChongLou> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_ChongLou> _hash = (HashMap<Integer, Table_ChongLou> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_ChongLou _values = new Table_ChongLou();
 _values.m_CopyDrop [ 0 ] =  Integer.parseInt((String)valuesList.get(ID_COPY_DROP_1));
_values.m_CopyDrop [ 1 ] =  Integer.parseInt((String)valuesList.get(ID_COPY_DROP_2));
_values.m_CopyNum [ 0 ] =  Integer.parseInt((String)valuesList.get(ID_COPY_NUM_1));
_values.m_CopyNum [ 1 ] =  Integer.parseInt((String)valuesList.get(ID_COPY_NUM_2));
_values.m_Fight1 [ 0 ] =  Integer.parseInt((String)valuesList.get(ID_FIGHT1_0));
_values.m_Fight1 [ 1 ] =  Integer.parseInt((String)valuesList.get(ID_FIGHT1_1));
_values.m_Fight1 [ 2 ] =  Integer.parseInt((String)valuesList.get(ID_FIGHT1_2));
_values.m_Fight1 [ 3 ] =  Integer.parseInt((String)valuesList.get(ID_FIGHT1_3));
_values.m_Fight1 [ 4 ] =  Integer.parseInt((String)valuesList.get(ID_FIGHT1_4));
_values.m_Fight1 [ 5 ] =  Integer.parseInt((String)valuesList.get(ID_FIGHT1_5));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

File: TLBB1.5/Client/Assets/Client/Scripts/Table/Table_ChongLou.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_ChongLou : ITableOperate
{ private const string TAB_FILE_DATA = "ChongLou.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_FIGHT1_0,
ID_FIGHT1_1,
ID_FIGHT1_2,
ID_FIGHT1_3,
ID_FIGHT1_4,
ID_FIGHT1_5,
ID_COPY_DROP_1,
ID_COPY_NUM_1,
ID_COPY_DROP_2,
ID_COPY_NUM_2,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private int[] m_CopyDrop = new int[2];
 public int GetCopyDropbyIndex(int idx) {
 if(idx>=0 && idx<2) return m_CopyDrop[idx];
 return -1;
 }

private int[] m_CopyNum = new int[2];
 public int GetCopyNumbyIndex(int idx) {
 if(idx>=0 && idx<2) return m_CopyNum[idx];
 return -1;
 }

private int[] m_Fight1 = new int[6];
 public int GetFight1byIndex(int idx) {
 if(idx>=0 && idx<6) return m_Fight1[idx];
 return -1;
 }

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_ChongLou _values = new Tab_ChongLou();
 _values.m_CopyDrop [ 0 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_COPY_DROP_1] as string);
_values.m_CopyDrop [ 1 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_COPY_DROP_2] as string);
_values.m_CopyNum [ 0 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_COPY_NUM_1] as string);
_values.m_CopyNum [ 1 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_COPY_NUM_2] as string);
_values.m_Fight1 [ 0 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_FIGHT1_0] as string);
_values.m_Fight1 [ 1 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_FIGHT1_1] as string);
_values.m_Fight1 [ 2 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_FIGHT1_2] as string);
_values.m_Fight1 [ 3 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_FIGHT1_3] as string);
_values.m_Fight1 [ 4 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_FIGHT1_4] as string);
_values.m_Fight1 [ 5 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_FIGHT1_5] as string);

 _hash.Add(nKey,_values); }


}
}

